packages/@uppy/core/src/Plugin.js

```diff
@@ -114,19 +114,30 @@ module.exports = class Plugin {
         // so it could still be called even after uppy.removePlugin or uppy.close
         // hence the check
         if (!this.uppy.getPlugin(this.id)) return
-        this.el = preact.render(this.render(state), targetElement, this.el)
+        render(this.render(state), targetElement)
         this.afterUpdate()
       }
+
       this._updateUI = debounce(this.rerender)
 
       this.uppy.log(`Installing ${callerPluginName} to a DOM element '${target}'`)
 
-      // clear everything inside the target container
       if (this.opts.replaceTargetContent) {
-        targetElement.innerHTML = ''
+        // Although you could remove the child nodes using DOM APIs (container.innerHTML = ''),
+        // this isn't recommended because the component might need to do additional cleanup when it is removed.
+        // To remove the rendered content and run any cleanup processes, render an empty element into the container:
+        render(h(null), targetElement)
       }
 
-      this.el = preact.render(this.render(this.uppy.getState()), targetElement)
+      // Since preact X the render function does not return a reference to the created element anymore.
+      // This is because it can sometimes return multiple elements now, likely due to fragments.
+      // To still get a reference in order to place it in `this.el`, we create a clone with a ref.
+      const ref = createRef()
+      const clone = cloneElement(this.render(this.uppy.getState()), { ref })
+
+      render(clone, targetElement)
+
+      this.el = ref.current
```

Comment:
If the root element is a component, this ref could be populated with its instance. It might be more reliable to do something like:

```js
      preact.render(this.render(this.uppy.getState()), targetElement);
      this.el = targetElement.firstElementChild;
```